daemon/policy.go

```diff
@@ -188,6 +188,9 @@ type PolicyAddEvent struct {
 
 // Handle implements pkg/eventqueue/EventHandler interface.
 func (p *PolicyAddEvent) Handle(res chan interface{}) {
+	if !p.d.dnsRuleGen.AreFQDNRulesStillPresent(p.rules) {
```

Comment:
I guess it should be ok because the queue ensures things are done one-at-a-time but I was expecting this check to happen in `policyAdd` after the policy repository lock was held. 
Wherever we do it, we should include a debug print here saying that this is a no-op. We also need to make sure that we don't need to handle returning a revision number or error (I'm not sure what the new code assumes, it may always wait for a response).

Reply:
Around return, etc.. all are ok :-)

About the place, for me is more in the handle, something that you want to check before you add the rules in policyAdd and avoid logs, etc.. but I'm open here.